src/main/resources/migrations/common/V2024.04.22__baseline_tables.sql

```diff
@@ -0,0 +1,328 @@
+--
+-- TABLES
+--
+
+---------------------------------------------------------------------------------------
+-- Contains coded reference values, used to constrain the values of lists/validation.
+-- e.g. Booking types, meeting types, hearing types, status etc..
+----------------------------------------------------------------------------------------
+
+CREATE TABLE reference_code
+(
+    reference_code_id   bigserial NOT NULL CONSTRAINT reference_code_pk PRIMARY KEY,
+    group_code          varchar(40) NOT NULL,  
+    code                varchar(40) NOT NULL,
+    description         varchar(100) NOT NULL,
+    created_by          varchar(100) NOT NULL,
+    created_time        timestamp NOT NULL,
+    amended_by          varchar(100),
+    amended_time        timestamp           
+);
+
+CREATE UNIQUE INDEX idx_reference_code_group ON reference_code(group_code, code);
+
+---------------------------------------------------------------------------------------
+-- These are the courts that are known to the service.
+-- A court has contacts who will be notified of bookings and changes.
+-- A court has users who have selected to view/add bookings for it.
+-- A court has video_bookings related to it.
+----------------------------------------------------------------------------------------
+
+CREATE TABLE court
+(
+    court_id            bigserial NOT NULL CONSTRAINT court_id_pk PRIMARY KEY,
+    code                varchar(30) NOT NULL UNIQUE,
+    description         varchar(100) NOT NULL,
+    enabled             boolean NOT NULL,
+    notes               varchar(200),
+    created_by          varchar(100) NOT NULL,
+    created_time        timestamp NOT NULL,
+    amended_by          varchar(100),
+    amended_time        timestamp 
+);
+
+CREATE INDEX idx_court_code ON court(code);
+
+---------------------------------------------------------------------------------------
+-- The people involved in video link bookings at each court.
+-- This can be a single group email contact or a list of individual contacts.
+-- This list of contacts will be notified of bookings / changes for this court.
+----------------------------------------------------------------------------------------
+
+CREATE TABLE court_contact
```

Comment:
The contact tables all look very similar.  Do we need 3 tables?  We could have a contact type?

I realise this means reference id gotchas.

Reply:
I was thinking there might be more - police, parole - to come, and wondering whether the contact methods for them might change. I went separate just for clarity .. picturing 2-3 lists related to a booking of the contacts to keep informed about it. Good challenge, but one I did think about and went for separate (for now!) ..